test/index.test.ts

```diff
@@ -45,9 +98,42 @@ describe("listhen", () => {
     expect(listener.url.endsWith("/foo/bar")).toBe(true);
     // eslint-disable-next-line no-console
     // expect(console.log).toHaveBeenCalledWith(expect.stringMatching('\n  > Local:    http://localhost:3000/foo/bar'))
+    const response = (await sendRequest(listener.url)) as string;
+    expect(JSON.parse(response)).toEqual({
+      path: "/foo/bar",
+      httpVersion: "1.1",
+    });
+  });
+
+  // see https://http2.github.io/faq/#does-http2-require-encryption
+  test("listen (http2)", async () => {
```

Comment:
This test throws the 2 unhandled errors, though all tests - including this one - are successfully executed. I don't know how to handle this. The test is not needed, but will show, that a http2 server is throwing an error (`Protocol Error`) when requesting a resource without encryption

Reply:
The CI check shows another error (Promise resolves instead of rejects). Locally the promise rejects, but the 2 errors still listed at the end of the test run

Reply:
[This](https://github.com/vitest-dev/vitest/issues/2262) (closed) issue describes the async throws error. 

Now the assertion is like in [this comment](https://github.com/vitest-dev/vitest/issues/2262#issuecomment-1301822351), means: No extra anonymous arrow function as argument to `expect`, an `await` in front of it and behind a [`rejects`](https://vitest.dev/api/expect.html#rejects), since `sendHttp2Request` returns a promise and is therefore an `async` function.

I've also tried to catch the error manually - without success.